storage/txcache/evictionStrategy.go

```diff
@@ -0,0 +1,168 @@
+package txcache
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/logger"
+)
+
+var log = logger.GetOrCreate("txcache/eviction")
+
+// EvictionStrategyConfig is a cache eviction model
+type EvictionStrategyConfig struct {
+	CountThreshold                 int
+	CountJustAFewSenders           int
+	NoOldestSendersToEvict         int
+	ALotOfTransactionsForASender   int
+	NoTxsToEvictForASenderWithALot int
+}
+
+// EvictionStrategy is a cache eviction model
+type EvictionStrategy struct {
+	Cache  *TxCache
+	Config EvictionStrategyConfig
+	mutex  sync.Mutex
+}
+
+// NewEvictionStrategy creates a new EvictionModel
+func NewEvictionStrategy(cache *TxCache, config EvictionStrategyConfig) *EvictionStrategy {
```

Comment:
instead of a type, you should input an interface.

Reply:
The `EvictionStrategy` is actually very coupled to `TxCache`'s implementation, because it knows a lot about its internal structures: `txByHash` and `txListBySender`. I will try to find a better design.

Reply:
Redesigned coupling of cache and eviction strategy - eviction methods are now methods of the cache, but in a separate file - `eviction.go`.

@sasurobert, is it better?